### PR TITLE
:sparkles: Split the multiclusterhub controller from global hub

### PR DESCRIPTION
**Pull request by yanmxa**
## Summary

encounter the error like
```bash
2024-11-05T07:07:20.920Z	ERROR	controller/controller.go:316	Reconciler error	{"controller": "transport", "controllerGroup": "operator.open-cluster-management.io", "controllerKind": "MulticlusterGlobalHub", "MulticlusterGlobalHub": {"name":"multicluster-global-hub-transport","namespace":"namespace-7g4m7k"}, "namespace": "namespace-7g4m7k", "name": "multicluster-global-hub-transport", "reconcileID": "2c3cbf7e-93dc-48aa-89b3-39789feb9e49", "error": "controller with name strimzi_controller already exists. Controller names must be unique to avoid multiple controllers reporting to the same metric"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.19.0/pkg/internal/controller/controller.go:316
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.19.0/pkg/internal/controller/controller.go:263
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).Start.func2.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.19.0/pkg/internal/controller/controller.go:224
```

fixed in commit: https://github.com/stolostron/multicluster-global-hub/pull/1194/commits/9cc0bfaa891155cf292812810488c2735fcc6795
## Related issue(s)

Fixes # https://issues.redhat.com/browse/ACM-15415

## Tests
* [x] Unit/function tests have been added and incorporated into `make unit-tests`.
* [x] Integration tests have been added and incorporated into `make integration-test`.
* [ ] E2E tests have been added and incorporated into `make e2e-test-all`.
* [ ] List other manual tests you have done.

### CODE REVIEW

**Comment by yanmxa** [comment]
environment error 

/retest

File: operator/pkg/controllers/multiclusterhub/acm.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	err := r.GetClient().Get(ctx, client.ObjectKeyFromObject(mch), mch)
+	if err != nil && errors.IsNotFound(err) {
+		config.SetACMResourceReady(false)
+		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+	} else if err != nil {
+		log.Errorf("failed to get the multiclusterhub: %v", err)
+		return ctrl.Result{}, err
+	}
+
+	if mch.Status.Phase != mchv1.HubRunning {
```
Comment (clyang82) [inline operator/pkg/controllers/multiclusterhub/acm.go:69]: @yanmxa We cannot trust the status for MCH. sometimes, the status is not running, but actually it works in most cases. we should not rely on Running status.
Let us continue checking the CRD exists or not.  FYI @ldpliu

**Pull request merged**